google/cloud/bigquery/table.py

```diff
@@ -234,9 +283,9 @@ def to_api_repr(self) -> dict:
             Dict[str, object]: Table reference represented as an API resource
         """
         return {
-            "projectId": self._project,
-            "datasetId": self._dataset_id,
-            "tableId": self._table_id,
+            "projectId": self.project,
+            "datasetId": self.dataset_id,
+            "tableId": self.table_id,
```

Comment:
Shouldn't this be a `copy.deepcopy` of `self._properties`?

Reply:
Fun fact - switching to the deepcopy implementation revealed a bug in the constructor. A string that was passed to the `_set_sub_prop()` helper resulted in a dysfunctional `_properties`  dict: `{"k": {"e": {"y": "value"}}}`, as that helper expects a sequence.